atlasdb-commons/src/test/java/com/palantir/common/proxy/ExperimentRunningProxyTest.java

```diff
@@ -0,0 +1,117 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.common.proxy;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.lang.reflect.Proxy;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.function.BooleanSupplier;
+import java.util.function.IntSupplier;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class ExperimentRunningProxyTest {
+    private static final int EXPERIMENTAL_INT = 123;
+    private static final int FALLBACK_INT = 1;
+    private static final RuntimeException RUNTIME_EXCEPTION = new RuntimeException("foo");
+
+    private final IntSupplier experimentalIntSupplier = mock(IntSupplier.class);
+    private final IntSupplier fallbackIntSupplier = mock(IntSupplier.class);
+    private final BooleanSupplier useExperimental = mock(BooleanSupplier.class);
+    private final Clock clock = mock(Clock.class);
+    private final ExperimentRunningProxy<IntSupplier> proxy = new ExperimentRunningProxy<>(
+            experimentalIntSupplier, fallbackIntSupplier, useExperimental, clock);
+
+    private final IntSupplier experimentSupplier = intSupplierForProxy(proxy);
+
+    @Before
+    public void setup() {
+        when(fallbackIntSupplier.getAsInt()).thenReturn(FALLBACK_INT);
+        when(experimentalIntSupplier.getAsInt()).thenReturn(EXPERIMENTAL_INT);
+        when(clock.instant()).thenReturn(Instant.ofEpochSecond(0));
+    }
+
+    @Test
+    public void doesNotAttemptExperimentIfNotRequested() {
+        disableExperiment();
+        assertThat(experimentSupplier.getAsInt()).isEqualTo(FALLBACK_INT);
+    }
+
+    @Test
+    public void attemptsExperimentIfRequested() {
+        enableExperiment();
+        assertThat(experimentSupplier.getAsInt()).isEqualTo(EXPERIMENTAL_INT);
+    }
+
+    @Test
+    public void canFallBackIfExperimentFails() {
+        enableExperiment();
+        throwOnFirstExperiment();
+
+        assertThatThrownBy(experimentSupplier::getAsInt).isEqualTo(RUNTIME_EXCEPTION);
+        assertThat(experimentSupplier.getAsInt()).isEqualTo(FALLBACK_INT);
+    }
+
+    @Test
+    public void attemptsExperimentAgainAfterEnoughTimeHasElapsed() {
+        enableExperiment();
+        throwOnFirstExperiment();
+
+        assertThatThrownBy(experimentSupplier::getAsInt).isEqualTo(RUNTIME_EXCEPTION);
+        assertThat(experimentSupplier.getAsInt()).isEqualTo(FALLBACK_INT);
+
+        when(clock.instant()).thenReturn(Instant.ofEpochSecond(1_000));
```

Comment:
why does this work (well I know why because I read the source)? I think you want to say something along the lines of:

```java
when(clock.instant()).thenReturn(Instant.ofEpochSecond(REFRESH_INTERVAL.minus(Duration.ofSecond(1))

// assert that it doesn't work

when(clock.instant()).thenReturn(Instant.ofEpochSecond(REFRESH_INTERVAL.plus(Duration.ofSecond(1))
```

if you don't want to refer to the constant, then just add it as a constructor parameter.

It's not clear from the test why adding a 1000 seconds in particular does anything, nor where any sort of refresh/backoff period comes into play.